definition-src/Game/LambdaHack/Core/Random.hs

```diff
@@ -23,12 +23,13 @@ import Game.LambdaHack.Core.Prelude
 import qualified Control.Monad.Trans.State.Strict as St
 import           Data.Ratio
 import qualified System.Random as R
+import qualified System.Random.SplitMix32 as SM
```

Comment:
I'm fine with this being 32bit everywhere, as long as it's not actually slower on amd64 systems than 64bit version.

Reply:
I (almost) never generate random `Word64`s. I generate mostly `Int`s, so I could just as well generate `Int32`s or `Word32`s, because that's what `Int` amounts to on GHCJS. So it seems 32bits is fine.

Reply:
https://hackage.haskell.org/package/random-1.1/docs/src/System.Random.html#line-345
```haskell
instance Random Int        where randomR = randomIvalIntegral; random = randomBounded
```

https://hackage.haskell.org/package/random-1.1/docs/src/System.Random.html#randomIvalIntegral
```haskell
-- The two integer functions below take an [inclusive,inclusive] range.
randomIvalIntegral :: (RandomGen g, Integral a) => (a, a) -> g -> (a, g)
randomIvalIntegral (l,h) = randomIvalInteger (toInteger l, toInteger h)
```

https://hackage.haskell.org/package/random-1.1/docs/src/System.Random.html#randomIvalInteger
```
randomIvalInteger :: (RandomGen g, Num a) => (Integer, Integer) -> g -> (a, g)
...
```